taskvine/src/bindings/python3/taskvine.binding.py

```diff
@@ -1574,25 +1581,99 @@ def submit(self, task):
+
+    # Turn a list of python functions into a Library
+    #
+    #
+    # @param self            Reference to the current manager object.
+    # @param name            Name of the Library to be created
+    # @param function_list   List of all functions to be included in the library
+    def create_library_from_functions(self, name, *function_list):
+        # ensure poncho python library is available
+        if poncho_available == False:
+            raise ModuleNotFoundError("The poncho module is not available. Cannot create Library.")
+        # positional arguments are the list of functions to include in the library
+        # create a unique hash of a combination of function names and bodies
+        functions_hash = package_serverize.generate_functions_hash(function_list)
+
+        # create path for caching library code and environment based on function hash
+        library_cache_path = f"{self.cache_directory}/vine-library-cache/{functions_hash}"
+        library_code_path = f"{library_cache_path}/library_code.py"
+        library_env_path = f"{library_cache_path}/library_env.tar.gz"
+
+        # library cache folder doesn't exist, create it
+        Path( library_cache_path ).mkdir(mode=0o755, parents=True, exist_ok=True)
+        # if the library code and environment exist, move on to creating the Library Task
+        if os.path.isfile(library_code_path) and os.path.isfile(library_env_path):
+            pass
+        else:
+            print("No cached Library code and environment found, regenerating...")
+            # create library code and environment
+            package_serverize.serverize_library_from_code(library_cache_path, function_list, name)
+            # enable correct permissions for library code
+            os.chmod(library_code_path, 0o775)
+        
+        # create Task to execute the Library
+        t = LibraryTask("python ./library_code.py", name)
+        # declare the environment
+        f = self.declare_poncho(self.declare_file(library_env_path))
+        t.add_environment(f)
+        # declare the library code as an input
+        f = self.declare_file(library_code_path)
+        t.add_input(f, "library_code.py")
+        return t
+
+    # Turn Library code created with poncho_package_serverize into a Library Task
+    #
+    #
+    # @param self            Reference to the current manager object.
+    # @param library_path    Path of the file which contains the library code - the output of poncho_package_serverize
+    # @param env_path        Path of the environment file which contains the environment capable of running the Library
+    def create_library_from_serverized_files(self, name, library_path, env_path):
+        if poncho_available == False:
+            raise ModuleNotFoundError("The poncho module is not available. Cannot create library.")
+        t = LibraryTask("python ./library_code.py", name)
+        f = self.declare_poncho(self.declare_file(env_path))
```

Comment:
With `declare_poncho` in python you can simply pass the tarball. Also, I'd allow the user to pass any environment that they want, like:
```python
...
# param env   Either a vine file or a poncho tarball filename that expands to an environment to run the library.

def create_library_from_serverized_files(self, name, library_path, env):
...
if env:
  if isinstance(env, str):
    env = self.declare_poncho(env, cache=True)
  t.add_environment(env)
```
Note that in the python bindings passing a str to declare_poncho does the right thing (i.e. declares a local file if needed.)